packages/dialog/src/vaadin-dialog.js

```diff
@@ -3,397 +3,15 @@
-      if (!this.footerContainer.parentElement) {
-        this.appendChild(this.footerContainer);
-      }
-
-      if (this.footerRenderer) {
-        // Only call header renderer after the container has been initialized
-        this.footerRenderer.call(this.owner, this.footerContainer, this.owner);
-      }
-    }
-
-    this._headerTitleRenderer();
-
-    this.__updateOverflow();
-  }
-
-  /**
-   * Updates the coordinates of the overlay.
-   * @param {!DialogOverlayBoundsParam} bounds
-   */
-  setBounds(bounds) {
-    const overlay = this.$.overlay;
-    const parsedBounds = { ...bounds };
-
-    if (overlay.style.position !== 'absolute') {
-      overlay.style.position = 'absolute';
-      this.setAttribute('has-bounds-set', '');
-      this.__forceSafariReflow();
-    }
-
-    Object.keys(parsedBounds).forEach((arg) => {
-      if (typeof parsedBounds[arg] === 'number') {
-        parsedBounds[arg] = `${parsedBounds[arg]}px`;
-      }
-    });
-
-    Object.assign(overlay.style, parsedBounds);
-  }
-
-  /**
-   * Retrieves the coordinates of the overlay.
-   * @return {!DialogOverlayBounds}
-   */
-  getBounds() {
-    const overlayBounds = this.$.overlay.getBoundingClientRect();
-    const containerBounds = this.getBoundingClientRect();
-    const top = overlayBounds.top - containerBounds.top;
-    const left = overlayBounds.left - containerBounds.left;
-    const width = overlayBounds.width;
-    const height = overlayBounds.height;
-    return { top, left, width, height };
-  }
-
-  /**
-   * Safari 13 renders overflowing elements incorrectly.
-   * This forces it to recalculate height.
-   * @private
-   */
-  __forceSafariReflow() {
-    const scrollPosition = this.$.resizerContainer.scrollTop;
-    const overlay = this.$.overlay;
-    overlay.style.display = 'block';
-
-    requestAnimationFrame(() => {
-      overlay.style.display = '';
-      this.$.resizerContainer.scrollTop = scrollPosition;
-    });
-  }
-
-  /** @private */
-  __updateOverflow() {
-    let overflow = '';
-
-    // Only set "overflow" attribute if the dialog has a header, title or footer.
-    // Check for state attributes as extending components might not use renderers.
-    if (this.hasAttribute('has-header') || this.hasAttribute('has-footer') || this.headerTitle) {
-      const content = this.$.content;
-
-      if (content.scrollTop > 0) {
-        overflow += ' top';
-      }
-
-      if (content.scrollTop < content.scrollHeight - content.clientHeight) {
-        overflow += ' bottom';
-      }
-    }
-
-    const value = overflow.trim();
-    if (value.length > 0 && this.getAttribute('overflow') !== value) {
-      this.setAttribute('overflow', value);
-    } else if (value.length === 0 && this.hasAttribute('overflow')) {
-      this.removeAttribute('overflow');
-    }
-  }
-}
-
-customElements.define(DialogOverlay.is, DialogOverlay);
+export { DialogOverlay } from './vaadin-dialog-overlay.js';
```

Comment:
Added re-export for backwards compatibility, as some addons might import the overlay from here.

Reply:
Should re-export be also done for TS?

Reply:
Yes, it's already done - see [this line](https://github.com/vaadin/web-components/pull/5307/files#diff-66cabdb292df25142260f07d3fd14c092307694ee525234518d64ca16c513600R11).